### ISSUE TITLE
[Jupyter] No error messages when doing something in the napari Window

### ISSUE THREAD

**Issue by jwrth** [napari/napari#6465]
### 🐛 Bug Report

I have the problem that if there is an error raised while doing something inside the napari window, I do not see this anywhere. Shouldn't there be some kind of window popping up and giving me information on the error? Or printing it to the console?

### 💡 Steps to Reproduce

Here is some code to illustrate my problem:
```python
import napari
import numpy as np
from napari.layers import Image
from magicgui import magicgui

@magicgui(image={'label': 'Pick an Image'})
def my_widget(image: Image):
    raise ValueError("Error")
    print('hello')

viewer = napari.view_image(np.random.rand(64, 64), name="My Image")
viewer.window.add_dock_widget(my_widget)
``` 

When pressing the "Run" button a ValueError should be raised. But instead nothing happens. If I comment out the ValueError, it prints "hello" to the console as expected. Is there a way to catch errors and display them somehow? This would make debugging a lot easier.

Thanks a lot in advance for your help!

### 🌎 Environment

napari: 0.4.18
Platform: Windows-10-10.0.19045-SP0
Python: 3.9.18 (main, Sep 11 2023, 14:09:26) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.25.2
SciPy: 1.9.3
Dask: 2023.9.2
VisPy: 0.12.2
magicgui: 0.7.3
superqt: 0.6.0
in-n-out: 0.1.8
app-model: 0.2.2
npe2: 0.7.2

OpenGL:
  - GL version:  4.6.0 Compatibility Profile Context 23.10.24.05.230830
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 2560x1600, scale 1.0
  - screen 2: resolution 3840x2160, scale 1.0
  - screen 3: resolution 2880x1800, scale 1.0

Settings path:
  - AppData\Local\napari\xparse_57fe1a37b30a9e37de3a06866d324c7f56a92d1a\settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.0.8 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)
  - ome-types: 0.4.2 (2 contributions)

**Comment by Czaki** [comment]
There should be dialog poping in the bottom right corner of the canvas. They disappear after some time.

**Comment by Czaki** [comment]
Yes. Popup will not show if you create a napari viewer from Jupyter notebook as we do not install a custom hook for exceptions in such cases. 

But I do not know why it is not printed in notebook.

**Comment by jwrth** [comment]
Ok good to know that this is the expected behaviour! Thanks!

**Comment by psobolewskiPhD** [comment]
This issue is like a worse version of https://github.com/napari/napari/issues/6455
Same root cause I suspect.

**Comment by Czaki** [comment]
No. this is a lighter version. As in #6455 message is not printed anywhere, and here it looks like a problem with Jupyter server as default hooks should be called.

**Comment by jwrth** [comment]
Is there a workaround for this or currently not?

**Comment by Czaki** [comment]
Try this code at the beginning of notebook:

```python
import sys 
import traceback

def my_excepthook(type_, value, trace_back):
	traceback.print_exception(type, value, trace_back, file=sys.stderr)

sys.excepthook = my_excepthook
```

**Comment by jwrth** [comment]
Unfortunately, I still cannot see the error messages.

**Comment by Czaki** [comment]
What if you change `sys.stderr` to `sys.stduot`?

**Comment by jwrth** [comment]
No this also does not fix it...

**Comment by Chris-N-K** [comment]
I can actually not reproduce the error when running jupyter alone or through VS code.
In all configurations (pure jupyter notebook, VS code  ipynb with python env, VS code ipynb with jupyter kernel) do not get the popup, what's not optimal, but I do get an error traceback in jupyter.

My best guess is there might be something besides napari play into this issue.

Did you try running the notebook through `jupyter notebook` directly?
Do you get the same result when doing it this way?

**Comment by psobolewskiPhD** [comment]
@jwrth @Chris-N-K 
I believe that my PR: https://github.com/napari/napari/pull/6882 should have solved this.
The notification in the viewer should happen now.

**Comment by psobolewskiPhD** [comment]
Grrr!
The example above still doesn't result in a notification with the traceback in iPython and JupyterLab -- it does it napari is run stand-alone.
